File: solutions/scala/problem-00006.scala
/*
[minhuang@mocha:scala]$ time scala problem-00006.scala 
25164150

real  0m1.280s
user  0m0.232s
sys 0m0.044s
*/
val asq = (1 to 100).map(i => i * i).reduceLeft(_ + _)
val b = (1 to 100).reduceLeft(_ + _)
println(b*b - asq)

File: solutions/haskell/problem-00006.hs
{-
[minhuang@mocha:haskell]$ time runghc problem-00006.hs 
25164150

real    0m0.342s
user    0m0.304s
sys     0m0.016s
-}
result = (sum [1..100])^2 - (sum [x^2 | x <- [1..100]])
main = print result
